src/foam/nanos/auth/PasswordRemovalSink.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.auth;
+
+import foam.core.Detachable;
+import foam.core.FObject;
+import foam.core.X;
+import foam.dao.ProxySink;
+import foam.dao.Sink;
+
+public class PasswordRemovalSink
+    extends ProxySink
+{
+  public PasswordRemovalSink(X x, Sink delegate) {
+    setX(x);
+    setDelegate(delegate);
+  }
+
+  @Override
+  public void put(FObject obj, Detachable sub) {
+    FObject clone = obj.fclone();
+    clone.setProperty("password", null);
```

Comment:
Next three lines could be in a static method which could also be called from UserAndGroupAuthService to avoid duplicating the same logic. Or vise-versa.

Reply:
Done

Reply:
You should call that static method from here.

Reply:
Done for real this time.